app/models/project.rb

```diff
@@ -56,6 +69,24 @@ class Project < ActiveRecord::Base
     scope
   }
 
+  # Methods used to negotiate temporary abstract attribute `docker_build_method`
+  def docker_build_method
+    result = DOCKER_BUILD_METHODS.detect do |build_method|
+      build_method[:method] != 'samson' && send("#{build_method[:method]}?")
+    end
+
+    result&.[](:method) || 'samson'
+  end
+
+  def docker_build_method=(selected_method)
+    deselected_methods = DOCKER_BUILD_METHODS.reject { |bm| bm.include?(selected_method) }
```

Comment:
how is this include supposed to work on a hash ? ... missing test ?

Reply:
Tests passed because it would just set all methods to false, then set the one `selected_method` to true. I don't think we really need `deselected_methods` since it only saves one assignment.